src/sentry/static/sentry/app/views/organizationStream/createSavedSearchButton.jsx

```diff
@@ -87,20 +87,24 @@ class CreateSavedSearchButton extends React.Component {
 
   render() {
     const {isSaving, isModalOpen, error} = this.state;
-    const {organization, query} = this.props;
+    const {organization, query, children} = this.props;
 
     return (
       <Access organization={organization} access={['org:write']}>
-        <StyledButton
-          title={t('Add to organization filter list')}
-          size="zero"
-          borderless
-          type="button"
-          onClick={this.onToggle}
-          data-test-id="save-current-search"
-          aria-label={t('Add to organization filter list')}
-          icon="icon-add-to-list"
-        />
+        {children ? (
```

Comment:
I'm not totally sure what the right move it for something like this. calling `children` with `onToggle` as a param gives us unlimited flexibility, but also will be harder to reliably test. 

As an alternative, we could accept a classname as a prop that is applied to the button, but there will need to be more styles applied that way, since we switch to a base `<a>` on the dropdown menu version.

Reply:
I think using a render prop here is a bit overkill unless we needed a way to completely customize how the create saved search modal gets opened. Since it's mostly presentational customizations on the `Button` I would recommend the classname approach, or even a `buttonProps` where we just pass the entire object into `Button`, otherwise it looks like you're duplicating most of these props

Reply:
I held off on passing a class because I didn't want to have the onClick handler attached to a `span` element or whatever and mess up a bunch of the styling.

But the `buttonProps` idea is very compelling because I can merge them with `onClick`.

Reply:
i mixed up my two ideas here and this last comment makes no sense. ignore me.